### PR TITLE
Add aptkey=False to use pkgrepo.managed without apt-key

### LINKED ISSUE #59785: [FEATURE REQUEST] pkgrepo should follow current Debian standards
**Is your feature request related to a problem? Please describe.**
Debian have deprecated `apt-key` and strongly recommended no longer adding third party keys to the global apt trust.
Instead, these recommendations should be followed: https://wiki.debian.org/DebianRepository/UseThirdParty

> The key MUST be downloaded over a secure mechanism like HTTPS to a location only writable by root, which SHOULD be `/usr/share/keyrings`. The key MUST NOT be placed in `/etc/apt/trusted.gpg.d` or loaded by `apt-key add`.
>
> A sources.list entry SHOULD have the signed-by option set. The signed-by entry MUST point to a file, and not a fingerprint.

Note that the name of the key file `NAME-archive-keyring.gpg` and sources list `NAME.list` should match.

**Describe the solution you'd like**
When the `pkgrepo.managed` state should be able (and ideally prefer) to set up repos according to these specifications.
```sls
deb [arch=amd64 signed-by=/usr/share/keyrings/salt-archive-keyring.gpg] http://repo.saltproject.io/py3/debian/10/amd64/latest buster main:
  pkrepo.managed:
    - file: /etc/apt/sources.list.d/salt.list
    - key_url: https://repo.saltproject.io/py3/ubuntu/20.04/amd64/latest/salt-archive-keyring.gpg
```

The same should be done when the key is from a keyserver. The file may also require de-armouring if the repo isn't following the new specification.

To make this easier, the current (inconsistent with other backends) behaviour of requiring the `name` to be the verbatim line may need deprecating or otherwise avoiding. For standards-compliant repos it should be as simple as possible, with `file`, `key_url`, `comps` using the defaults.

```sls
salt-repo:
  pkgrepo.managed:
    - architectures: amd64
    - baseurl: http://repo.saltproject.io/py3/debian/10/amd64/latest  # would ideally default to name
    - dist: {{ grains["oscodename"] }}
```

**Describe alternatives you've considered**
Doing it manually with `file.managed` states and ignoring `pkgrepo`.

**Additional context**
Salt itself echoes this specification in its instructions: https://repo.saltproject.io/#debian
It's a bit paradoxical that it doesn't have the ability to follow it itself.

### PR DESCRIPTION
**Pull request by Ch3LL**
### What does this PR do?
Allows users to not use apt-key. Apt-key is on a deprecation path right now so this PR adds the ability to use `aptkey: False` with the pkgrepo.managed state to not use the `apt-key` cli command. Also if `apt-key` does not exist `apt-key is not used automatically.

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/59785

### CODE REVIEW

File: salt/states/pkgrepo.py [hunk 1]
```diff
@@ -307,7 +307,15 @@ def managed(name, ppa=None, copr=None, **kwargs):
        :mod:`pkg.latest <salt.states.pkg.latest>` to trigger the
        running of ``apt-get update`` prior to attempting to install these
        packages. Setting a require in the pkg state will not work for this.
+
+    aptkey: Use the binary apt-key
+
+    signedby:
+        On apt-based systems, ``signedby`` is the the path to the key file
+        the repository will use. This is required in apt-key is False.
```
Comment (OrangeDog) [inline salt/states/pkgrepo.py:315]: required **if** apt-key

File: salt/states/pkgrepo.py [hunk 2]
```diff
@@ -307,7 +307,15 @@ def managed(name, ppa=None, copr=None, **kwargs):
        :mod:`pkg.latest <salt.states.pkg.latest>` to trigger the
        running of ``apt-get update`` prior to attempting to install these
        packages. Setting a require in the pkg state will not work for this.
+
+    aptkey: Use the binary apt-key
+
+    signedby:
```
Comment (OrangeDog) [inline salt/states/pkgrepo.py:313]: How does this interact with `name`?
Can you put it in the name instead?
Does this override the one in the name if present?

**Comment by Ch3LL** [comment]
re-run all

**Review by twangboy — DISMISSED** [review]

**Review by MKLeb — DISMISSED** [review]

File: salt/states/pkgrepo.py [hunk 3]
```diff
@@ -307,7 +370,15 @@ def managed(name, ppa=None, copr=None, **kwargs):
        :mod:`pkg.latest <salt.states.pkg.latest>` to trigger the
        running of ``apt-get update`` prior to attempting to install these
        packages. Setting a require in the pkg state will not work for this.
+
+    aptkey: Use the binary apt-key
```
Comment (waynew) [inline salt/states/pkgrepo.py:374]: *question* would be appropriate to note that this is ignored if apt-key is not found on the path?

File: salt/modules/aptpkg.py [hunk 4]
```diff
@@ -136,7 +138,7 @@ def __init__(self, line, file=None):
             self.uri = ""
             self.line = line
             self.architectures = []
-            self.signed_by = ""
+            self.signedby = ""
```
Comment (waynew) [inline salt/modules/aptpkg.py:141]: **question** why the rename? Was the previous name buggy? I'm on my phone ATM so can't easily search for uses of this within salt - just want to make sure the rename doesn't catch anything

**Review by waynew — CHANGES_REQUESTED** [review]
just a few questions, otherwise from just looking on my phone it looks good

**Review by waynew — APPROVED** [review]

**Pull request merged**